byway/src/main/webapp/destinations.js

```diff
@@ -218,21 +237,50 @@ function getCurrentAddress(){
 * add event listener for submit button
 */
 window.onload = function(){
+  
   let nextButton = document.getElementById('next-button');
   nextButton.addEventListener('click', () => {
-      nextButton.href = buildUrlWithParams("/interests.html", {tripKey});
+      nextButton.href = configureTripKeyForPath(tripKey, '/interests.html');
   });
   document.getElementById('user-input-form').addEventListener('submit', (event) => {
     event.preventDefault();
-    const formData = new  FormData(document.getElementById("user-input-form"));
-    fetch(buildUrlWithParams("/api/destinations", {tripKey}), {method: 'POST', body:formData}).then((response)=>
-        response.json()).then(locationData => {
-        updateLocations(locationData);
-        updateStartDestination(locationData);
-      });
-    });
+    sendPlaceId("destinations-search-box");
+    sendPlaceId("start-search-box");
+  });
 }
 
+/** 
+* gets User Input from search box and sends the Place Id of user input to the server
+* elementName indicates which search box to get from, param indicates which param to save as
+* @param {String} elementName
+*/
+function sendPlaceId(elementName){
+  let formData = new FormData();
+  const request = {
+    query: String(document.getElementById(elementName).value),
+    fields: ["place_id"]
+  };
+  service.findPlaceFromQuery(request, (results, status) => {
+    console.log(status);
+    if (status === google.maps.places.PlacesServiceStatus.OK) {
+      formData.append(elementName, results[0].place_id);
+      fetch(configureTripKeyForPath(tripKey, '/api/destinations'), {method: 'POST', body:formData})
+      .then((response)=>
+        response.json())
+      .then(locationData => {
+        if (elementName == "start-search-box"){
```

Comment:
Seems like this is separate logic from the actual call to the service. It would be better to move this outside this function and separate it from the code that makes requests to the server. This could be moved to a separate function, or you could otherwise return a promise and handle the updating the UI in this functions caller.

Reply:
moved to seperate function

Reply:
this function actually ended up being unnecessary so i removed it